### PR TITLE
Switch from requests to got

### PR DESCRIPTION
**Pull request by hunterwerlla**
- Switch to got
- As a follow up later, we need to split up `HttpResourceFetcher`. It doesn't make sense in it's current form and doesn't report errors for streams (in the one place we use it, we fail later. This issue is in the existing version as well)

## Testing
- Made sure download lambda works
- Made sure download resources works
- Tested failures using invalid urls

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/resourcefetcher/httpResourceFetcher.ts [hunk 1]
```diff
@@ -26,19 +28,25 @@ export class HttpResourceFetcher implements ResourceFetcher {
...
+     * Returns the contents of the resource, or undefined if the resource could not be retrieved or if it
+     * is piped somewhere else
      */
     public async get(): Promise<string | undefined> {
         try {
             this.logger.verbose(`Loading ${this.logText()}`)
 
-            const contents = (await this.getResponseFromGetRequest()).body
-
+            let response: Response<string> | undefined = undefined
```
Comment (bryceitoc9) [inline src/shared/resourcefetcher/httpResourceFetcher.ts:38]: Nit: unnecessary `= undefined`

**Review by bryceitoc9 — APPROVED** [review]
LGTM but you may want to just move the piping in this PR before we forget about it.

File: src/shared/resourcefetcher/httpResourceFetcher.ts (hunk 1 above)
Comment (hunterwerlla) [reply src/shared/resourcefetcher/httpResourceFetcher.ts]: Removed

**Review by bryceitoc9 — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -1409,7 +1409,6 @@
         "@types/mocha": "^7.0.2",
         "@types/node": "^10.14.22",
         "@types/readline-sync": "^1.4.3",
-        "@types/request": "^2.47.1",
```
Comment (justinmk3) [inline package.json:1412]: is there a `@types/got` ?

**Review by justinmk3 — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (hunterwerlla) [reply package.json:1412]: Got is written in typescript :D

**Pull request merged**